plugins/meta/dnsname/files.go

```diff
@@ -145,6 +145,84 @@ func appendToFile(path, podname string, aliases []string, ips []*net.IPNet) erro
 	return nil
 }
 
+func removeHostLinesByIP(path string, ips []*net.IPNet) (bool, error) {
+	f, err := os.OpenFile(path, os.O_RDWR, 0o644)
+	if err != nil {
+		return false, err
+	}
+
+	defer func() {
+		if err := f.Close(); err != nil {
+			logrus.Errorf("failed to close file %q: %v", path, err)
```

Comment:
"Failed to close...". You can return error in this case.

Reply:
Done